pkg/resource/replication_group/custom_update_api.go

```diff
@@ -71,27 +71,9 @@ func (rm *resourceManager) CustomModifyReplicationGroup(
 	}
 
 	// Order of operations when diffs map to multiple updates APIs:
-	// 1. When automaticFailoverEnabled differs:
-	//		if automaticFailoverEnabled == false; do nothing in this custom logic, let the modify execute first.
-	// 		else if automaticFailoverEnabled == true then following logic should execute first.
-	// 2. When multiAZ differs
-	// 		if multiAZ = true  then below is fine.
-	// 		else if multiAZ = false ; do nothing in custom logic, let the modify execute.
```

Comment:
> When desired value is false then invoke modifyReplicationGroup (default implementation in sdk.go) first

`modifyReplicationGroup` only leads to the default/generated implementation if it returns `nil, nil`, which happens when there is no difference in security group IDs or engine version. That is to say, if we want to skip straight to the generated code, we should immediately `return nil, nil` instead of call `rm.modifyReplicationGroup`. 

> When desired value is true then let the rest of the custom modify logic execute first for AF/multiAZ to be applied later once all custom logic has executed.

This second case is equivalent to removing the below block, so the uncertainty remains with the first case above. Which of the below fields require multi AZ and auto failover to be disabled first (replica count is one, but I'm not sure if there's another)? If order matters then we should have had a test which failed upon removing this block, but we can add one now if it doesn't exist.

Reply:
Went ahead and reverted the changes to this particular file. Since we have limited time to iterate on this (as discussed offline), we can hold these thoughts for later. Same with non-blocking comments below, let's merge if this test run passes.